File: Graphs/adjacency_matrix/dfsAM.py
def dfs(graph, start_node, values, seen):
    seen.add(start_node)
    values.append(start_node)
    for i, val in enumerate(graph[start_node]):
        if val > 0 and i not in seen:
            dfs(graph, i, values, seen)


graph = [
    [0,1,0,1,0,0,0,0,0],
    [1,0,0,0,0,0,0,0,0],
    [0,0,0,1,0,0,0,0,1],
    [1,0,1,0,1,1,0,0,0],
    [0,0,0,1,0,0,1,0,0],
    [0,0,0,1,0,0,0,0,0],
    [0,0,0,0,1,0,0,1,0],
    [0,0,0,0,0,0,1,0,0],
    [0,0,1,0,0,0,0,0,0]
]
values = []
seen = set()
dfs(graph, 0, values, seen)
print(values)

File: Graphs/adjacency_matrix/dfsAM.js
const dfs = function (graph, start_node, values, seen) {
  seen.add(start_node);
  values.push(start_node);
  for (let i=0;i < graph[start_node].length;i++) {
    value = graph[start_node][i];
    if (value > 0 && !seen.has(i)) {
      dfs(graph, i, values, seen);
    }
  }
};

const graph = [
  [0, 1, 0, 1, 0, 0, 0, 0, 0],
  [1, 0, 0, 0, 0, 0, 0, 0, 0],
  [0, 0, 0, 1, 0, 0, 0, 0, 1],
  [1, 0, 1, 0, 1, 1, 0, 0, 0],
  [0, 0, 0, 1, 0, 0, 1, 0, 0],
  [0, 0, 0, 1, 0, 0, 0, 0, 0],
  [0, 0, 0, 0, 1, 0, 0, 1, 0],
  [0, 0, 0, 0, 0, 0, 1, 0, 0],
  [0, 0, 1, 0, 0, 0, 0, 0, 0],
];
const seen = new Set();
const values = [];
dfs(graph, 0, values, seen);
console.log(values);
